telegram/_message.py

```diff
@@ -3675,6 +3693,12 @@ def _parse_markdown(
                             "Spoiler entities are not supported for Markdown version 1"
                         )
                     insert = f"||{escaped_text}||"
+                elif entity.type == MessageEntity.BLOCKQUOTE:
+                    if version == 1:
+                        raise ValueError(
+                            "Blockquote entities are not supported for Markdown version 1"
+                        )
+                    insert = f">{escaped_text}"
```

Comment:
needs multiline handling as already being discussed in https://t.me/pythontelegrambotdev/965. Please update the tests to cover these cases :)

Reply:
Sorry for the late reply.
So, I came up with this, which I think would fit most cases

```python
insert = "\n>" + escaped_text.replace("\n", "\n>") + "\n"
```
I'm assuming that the output of `_parse_(html, markdown*)` should produce the same visual structure as the original message if/when used like `message.reply_text(message.text_markdown/html)`

these are the reasons for the extra `\n`s
- each quote line/paragraph, including the first, must be in its own new line. So trying to send `">line 1>line2"` or `"abc >def"` as text with `pars_mode=mdv2`, will raise `telegram.error.BadRequest: Can't parse entities: character '>' ...`
- the parsed quote must be separated from the following entities/text with a line break . This is to prevent the last line of the quote from including the rest of the message text as part of the quote.

Issues am facing is when the message starts (or ends) with a blockquote entity. There will be an extra unneeded blank lines.
I guess I can trim them at `message.(text/caption)_(html/markdown)*`, but don't like the repetition. would appreciate any ideas, or even a completely different approach

Reply:
Not to worry. Thanks for your elaborations! I haven't looked into it deeply yet, but a spontanuous idea would be that one should be able to use `MessageEntity.{offset, length}` to determine if the quoted text is at the start/end of the message. Also I'm wondering if `escaped_text.replace("\n", "\n>")` works as intended if the quote-text itself contains "\n", i.e. something like

> This is \n a quote

🤔

Reply:
> one should be able to use `MessageEntity.{offset, length} to ...`

Oh nice, problem solved then.

> Also I'm wondering if `escaped_text.replace("\n", "\n>")` works as intended

Hmm, apparently it does. Telegram escapes `\n`s when its meant literally. and surprisingly `string.replace("\n", "...")` will not match escaped `\\n`s

```bash
$ python
Python 3.11.4 ...
>>> """line 1
... line \\n 2
... line 3""".replace("\n", '  **  ')
'line 1  **  line \\n 2  **  line 3'
>>>
```

Reply:
okay, that's good, then :) adding this case to the tests would still be sane IMO :D